app/scripts/CenterTrack.jsx

```diff
@@ -170,13 +170,13 @@ class CenterTrack extends React.Component {
     );
   }
 
-  brushedX() {
+  brushedX(event) {
     // Need to reassign variable to check after reset
     const rangeSelectionMoved = this.rangeSelectionMoved;
     this.rangeSelectionMoved = false;
 
     if (
-      !this.sourceEvent ||
+      !sourceEvent(event) ||
       !this.props.onRangeSelectionX ||
```

Comment:
https://observablehq.com/@d3/d3v6-migration-guide

Reply:
hmm I'm a bit confused on this... `package.json` has older versions of d3 vite seems to be resolving to some later version. It would probably (actually be easier) if we merged your webpack changes first, and then rebased this PR on that.

Reply:
https://github.com/d3/d3-selection/releases/tag/v2.0.0